devicetypes/smartthings/qubino-flush-thermostat.src/qubino-flush-thermostat.groovy

```diff
@@ -0,0 +1,339 @@
+	log.warn "Unhandled command: ${cmd}"
+	[:]
+}
+
+def setThermostatMode(String mode) {
+	def modeValue = 0
+	if (state.supportedModes.contains(mode)) {
+		switch (mode) {
+			case "off":
+				modeValue = physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeSet.MODE_OFF
+				break
+			case "heat":
+				modeValue = physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeSet.MODE_HEAT
+				break
+			case "cool":
+				modeValue = physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeSet.MODE_COOL
+				break
+		}
+	} else {
+		log.debug "Unsupported mode ${mode}"
+	}
+
+	[
+			secure(zwave.thermostatModeV2.thermostatModeSet(mode: modeValue)),
+			"delay 2000",
+			secure(zwave.thermostatModeV2.thermostatModeGet())
+	]
+}
+
+def setTemperatureSetpoint(temperatureSetpoint) {
+	if (state.supportedModes.contains("heat")) {
+		sendHubCommand(setHeatingSetpoint(temperatureSetpoint))
+	} else {
+		sendHubCommand(setCoolingSetpoint(temperatureSetpoint))
+	}
+}
+
+def heat() {
+	setThermostatMode("heat")
+}
+
+def cool() {
+	setThermostatMode("cool")
+}
+
+def off() {
+	setThermostatMode("off")
+}
+
+def setHeatingSetpoint(setpoint) {
+	updateSetpoint(setpoint, physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointSet.SETPOINT_TYPE_HEATING_1)
+}
+
+def setCoolingSetpoint(setpoint) {
+	updateSetpoint(setpoint, physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointSet.SETPOINT_TYPE_COOLING_1)
+}
+
+def updateSetpoint(setpoint, setpointType) {
+	setpoint = temperatureScale == 'C' ? setpoint : fahrenheitToCelsius(setpoint)
+	setpoint = Math.max(Math.min(setpoint, maxSetpointTemperature), minSetpointTemperature)
+	[
+			secure(zwave.thermostatSetpointV2.thermostatSetpointSet([precision: 1, scale: 0, scaledValue: setpoint, setpointType: setpointType, size: 2])),
+			"delay 2000",
+			secure(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: setpointType))
+	]
+}
+
+def configure() {
+	secure(zwave.configurationV1.configurationGet(parameterNumber: 59))
+}
+
+def refresh() {
+	def cmds = [
+			secure(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: 1)),
+			secure(zwave.sensorMultilevelV5.sensorMultilevelGet()),
+			secure(zwave.thermostatModeV2.thermostatModeGet()),
+			secure(zwave.thermostatOperatingStateV2.thermostatOperatingStateGet()),
+			secure(zwave.thermostatSetpointV2.thermostatSetpointGet(setpointType: currentSetpointType)),
+			secure(zwave.meterV2.meterGet(scale: 0)),
+			secure(zwave.meterV2.meterGet(scale: 2))
+	]
+
+	delayBetween(cmds, 2500)
+}
+
+def ping() {
+	refresh()
+}
+
+def changeMode() {
+	if (state.supportedModes.contains("heat")) {
+		sendHubCommand(zwave.configurationV1.configurationSet(parameterNumber: 59, scaledConfigurationValue: 1))
+	} else {
+		sendHubCommand(zwave.configurationV1.configurationSet(parameterNumber: 59, scaledConfigurationValue: 0))
+	}
+}
+
+private secure(cmd) {
+	if (zwaveInfo.zw.endsWith("s")) {
```

Comment:
This needs to be `contains` now because of changes made to the zw object.

At some point I need to go back and update all the old DTHs as well.